src/analyz/ty_json.rs

```diff
@@ -1178,86 +1253,62 @@ fn try_render_ref_opty<'mir, 'tcx>(
     let d = icx.deref_operand(op_ty).unwrap();
     let is_mut = mutability == hir::Mutability::Mut;
 
+    let (prov, _offset) = d.ptr.into_parts();
+    let alloc = tcx.try_get_global_alloc(prov?)?;
+
+    let def_id = match alloc {
+        interpret::GlobalAlloc::Static(def_id) =>
+            def_id.to_json(mir),
+        interpret::GlobalAlloc::Memory(ca) => {
+            let ty = op_ty.layout.ty;
+            let def_id = match mir.allocs.get(ca, ty) {
+                Some(alloc_id) => alloc_id.to_owned(),
+                None => {
+                    // create the allocation
+                    let static_ref = make_static_ref(mir, icx, rty, d, is_mut, ca);
+                    mir.allocs.insert(tcx, ca, ty, static_ref)
+                }
+            };
+            def_id.to_json(mir)
+        }
+        _ =>
+            // Give up
+            return None
+    };
+
     if !is_mut {
         match *rty.kind() {
-            // Special case for &str
+            // Special cases for &str and &[T]
+            //
+            // These cases are the same apart from the kind label, and
+            // it's possible they could be folded together. But it
+            // seems like a bad idea to conflate string slices and
+            // ordinary slices unless/until we're sure it won't create
+            // complications, now or in the future.
```

Comment:
I have combined the cases in 8427bae08b75fc12089e5df775210de1711581c9.

(That further change sounds like maybe it would be a good idea, but I think it's beyond the scope of what I can bite off right now)